### DISCUSSION TITLE
Is it possible to "stream" data using Dagster ops?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17510] [Q&A] [answered]
Hi everyone, is it possible to "stream" data using Dagster ops? e.g.
```python
def op1():
    while True:
        query external API 
        yield result

def op2(continuous_stream_from_op1):
    for x in continuous_stream_from_op1:
        print(x)
```
where op1 continuously creates output, and op2 continuously receive input from op1? Or do we need to wait until op1 is completed, and then op2 receive all of the inputs in one go?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698744156655039?thread_ts=1698744156.655039&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
We currently don't support op to op streaming. Here's a tracking issue: https://github.com/dagster-io/dagster/issues/4263

However, you can stream data back to dagster from external API within a single op. You can do so using [Dagster Pipes](https://docs.dagster.io/guides/dagster-pipes). Here's a tutorial of invoking subprocess from dagster and streaming back info from that external process: https://docs.dagster.io/guides/dagster-pipes/subprocess

**Reply by erenz14** [reply]
Hi Yuhan,
 
I'm trying to do exactly what you mentioned Pipes can be used for: Have an op that runs a command in external code, and then stream some data back from the external code to the op.
 
However, after looking through all of the guides in the documentation and through the _PipesContext_ class, it looks like only assets allow for this functionality through _report_asset_materialization_.
 
I know you can use _context.log.info_ for logging messages, but is there any way to send something like a json object from the external code back to an op?

**Reply by iUnknownAdorn** [reply]
@erenz14 Did you ever find out an answer? 
My use case is to call LLM in one of the op, but not wait for its complete response. Based on the partial response start op_2.
